client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx

```diff
@@ -0,0 +1,388 @@
+import { useForm } from 'react-hook-form';
+import { useSelector } from 'react-redux';
+import React, { useContext } from 'react';
+import { ExpandMore } from '@material-ui/icons';
+import { Autocomplete } from '@material-ui/lab';
+import { differenceInYears, format } from 'date-fns';
+import { Accordion, AccordionDetails, AccordionSummary, Avatar, Checkbox, Divider, FormControl,
+         FormControlLabel, Grid, MenuItem, Radio, RadioGroup, Select, TextField, Typography } from '@material-ui/core';
+
+import City from 'models/City';
+import Toggle from 'commons/Toggle/Toggle';
+import ContactType from 'models/ContactType';
+import DatePick from 'commons/DatePick/DatePick';
+import StoreStateType from 'redux/storeStateType';
+import FormInput from 'commons/FormInput/FormInput';
+import InteractedContact from 'models/InteractedContact';
+import useContactQuestioning from './useContactQuestioning';
+import FamilyRelationship from 'models/enums/FamilyRelationship';
+import IdentificationTypes from 'models/enums/IdentificationTypes';
+import InteractedContactFields from 'models/enums/InteractedContact';
+import PrimaryButton from 'commons/Buttons/PrimaryButton/PrimaryButton';
+import { occupationsContext } from 'commons/Contexts/OccupationsContext';
+import { interactedContactsContext } from 'commons/Contexts/InteractedContactsContext';
+import AlphanumericTextField from 'commons/AlphanumericTextField/AlphanumericTextField';
+
+import useStyles from './ContactQuestioningStyles';
+import { ADDITIONAL_PHONE_LABEL, OCCUPATION_LABEL, RELEVANT_OCCUPATION_LABEL } from '../PersonalInfoTab/PersonalInfoTab';
+
+const ContactQuestioning: React.FC = (): JSX.Element => {
+    const classes = useStyles();
+    const { errors, setError, clearErrors } = useForm({});
+
+    const interactedContactsState = useContext(interactedContactsContext);
+    const { occupations } = useContext(occupationsContext);
+    const cities = useSelector<StoreStateType, Map<string, City>>(state => state.cities);
+    const contactTypes = useSelector<StoreStateType, Map<number, ContactType>>(state => state.contactTypes);
+
+    const [familyRelationships, setFamilyRelationships] = React.useState<FamilyRelationship[]>();
+    const [currentInteractedContact, setCurrentInteractedContact] = React.useState<InteractedContact>();
+
+    const { getAllRelationships, saveContact, updateInteractedContact, changeIdentificationType, openAccordion, updateNoResponse } = useContactQuestioning({ setFamilyRelationships, setCurrentInteractedContact, interactedContactsState });
+
+    React.useEffect(() => {
+        getAllRelationships();
+    },[]);
+
+    return (
+        <>
+            <Typography className={classes.title} variant='body1'><b>טופס תשאול מגעים ({interactedContactsState.interactedContacts.length})</b></Typography>
```

Comment:
Maybe split the whole component to the three parts it contains?
Would be a lot more readable and easier to manage

Reply:
בוצע